src/v/serde/serde.h

```diff
@@ -533,15 +530,12 @@ inline void write(iobuf& out, iobuf t) {
     out.append(t.share(0, t.size_bytes()));
 }
 
-template<typename T>
-void write(iobuf& out, T t) {
-    using Type = std::decay_t<T>;
+template<typename Clock, typename Duration>
+void write(iobuf&, std::chrono::time_point<Clock, Duration> t) {
     static_assert(
-      !is_chrono_time_point<Type>,
+      !is_chrono_time_point<decltype(t)>,
```

Comment:
That'll do!

We also have a few instances of:
```c++
template<typename T>
struct dependent_false : std::false_type {};

{
   static_assert(dependent_false<T>::value, "Type name not defined");
}
```

Reply:
That would be nice cleanup, thanks!

Reply:
I will create a separate PR for this.